packages/peregrine/src/Router/__tests__/resolveUnknownRoute.test.js

```diff
@@ -26,6 +67,59 @@ test('Preload path: resolves directly from preload element', async () => {
     });
 });
 
+test('returns NOTFOUND when offline and requested content is not in cache ', async () => {
+    navigator.onLine = false;
+
+    fetch.mockResponseOnce(JSON.stringify(mockManifest));
+    const res = await resolveUnknownRoute({
+        route: 'foo-bar.html',
+        apiBase: 'https://store.com',
+        __tmp_webpack_public_path__: 'https://dev-server.com/pub'
+    });
+
+    expect(res).toHaveProperty('id', NotFoundManifest.NotFound.rootChunkID);
+});
+
+test('stores response of urlResolver in cache', async () => {
+    fetch.mockResponseOnce(urlResolverRes('PRODUCT'));
+    fetch.mockResponseOnce(JSON.stringify(mockManifest));
+
+    const url = 'foo-bar.html';
+
+    await resolveUnknownRoute({
+        route: url,
+        apiBase: 'https://store.com',
```

Comment:
Probably safer to use `example.com` here or some other invalid URL just in case this sends odd requests to a real site.

Reply:
Good call -- just in case preloading ever breaks. :+1: